### ISSUE TITLE
uint, float32, arraybuffer

### ISSUE THREAD

**Issue by jamesdbrock** [purescript-contrib/governance#40]
I'd like to propose these libraries for inclusion into purescript-contrib:

* https://github.com/zaquest/purescript-uint
* https://github.com/athanclark/purescript-float32
* https://github.com/jacereda/purescript-arraybuffer

I depend on these libraries, they are old and reliable, and the owners' attentions seem to have attenuated. Actually, it seems to
me that @athanclark was maintaining all three of these libraries, but his Github activity dropped in September 2020.

__uint__ and __float32__ in particular define basic types which would really benefit the PureScript ecosystem.

I will commit to maintaining these libraries.

If these libraries are added, then I think @zaquest , @athanclark , and @jacereda should also be made members of purescript-contrib , so they have the option of resuming active control over their libraries.

Is this the list of current maintainers?
https://github.com/orgs/purescript-contrib/people

References:
https://github.com/purescript-contrib/governance/issues/2
https://github.com/zaquest/purescript-uint/pull/12
https://github.com/athanclark/purescript-float32/pull/3
https://github.com/jacereda/purescript-arraybuffer/pull/28

**Comment by JordanMartinez** [comment]
Just curious, but why not fork these repos under your name, republish under new names, and continue using those from there?

Are there any contrib libraries currently using these libs as dependencies?

**Comment by athanclark** [comment]
I recommend the transference of maintenance to @jamesdbrock, he seems dedicated and would contribute meaningfully to the ecosystem

**Comment by zaquest** [comment]
I don't mind transferring `purescript-uint` to `purescript-contrib` if it's of any interest for the community.
Meanwhile, @jamesdbrock I sent you an invite to the repository so you could merge your PR. If you need any assistance from me to publish it Pursuit, please, let me know.

**Comment by jacereda** [comment]
Same situation here with `purescript-arraybuffer`, I don't mind transferring it.

**Comment by thomashoneyman** [comment]
Given the basic / foundational nature of these libraries, your offer to maintain them, and the endorsements of each maintainer I’m interested in adding these to contrib.

@zaquest, @athanclark, @jacereda — are you wanting to transfer maintainership entirely, or do you want to remain maintainers alongside @jamesdbrock?

The process from here is to update the libraries according to the contrib guidelines and then transfer them in (adding folks to the organization who need to be). If you need help with that, @jamesdbrock, then we could transfer the libraries in and I can help update them from there.

**Comment by athanclark** [comment]
I may wish to recommend changes to my library in the future, but I'm not fit to be the primary maintainer of the library if it's this important to the ecosystem; I'm comfortable with completely transferring maintainership to @jamesdbrock

**Comment by thomashoneyman** [comment]
Also, just to clarify: I’ll have to discuss with other contrib members about adding these libraries, as it’s not my decision alone.

We may add one or two or all three; for ones we don’t add, you can transfer ownership to @jamesdbrock or add him as a collaborator on the repo as another possible solution — as @JordanMartinez mentioned before.

**Comment by zaquest** [comment]
I haven't been doing any PureScript programming in years, so I'd rather give up any control over the library. There're PRs if I'd like to suggest changes)

**Comment by jamesdbrock** [comment]
Thanks everyone for being so helpful about this.

@zaquest and @jacereda , I've accepted your collaboration invitations.

@thomashoneyman I'm still proposing to add all of the libraries and all of the maintainers which I listed in the top of this issue to purescript-contrib. I'll wait for your decision.

**Comment by JordanMartinez** [comment]
@thomashoneyman Here's my thoughts (and responses to my previous comments):

> Just curious, but why not fork these repos under your name, republish under new names, and continue using those from there?

These three repos provide foundational types for the JS backend that should continue to be supported. Including them in the `contrib`'s repos makes it more likely that these repos will continue to be maintained over time.

> Are there any contrib libraries currently using these libs as dependencies?

I think we should follow the `core` libs policy of not allowing any repos into `core` if they depend on anything outside of `core`. It helps significantly when we update the ecosystem for PS releases with breaking changes. AFAICT, none of these repos seem to depend on non-core/non-contrib repos. 

In short, I'm in favor of adding these libraries due to the 'maintenance' concern above. Since they are small, I don't think it's a huge burden to add them to the repos, and I think most of them are already mature, so any new work on them should be minimal.

The only other question I have is what rights/privileges/roles/etc. should be granted to the original author and @jamesdbrock for each repo. It sounds like most would be fine with transferring it to James and submitting PRs to change anything if they wanted to.

**Comment by jamesdbrock** [comment]
Thanks @JordanMartinez

**Comment by zaquest** [comment]
I suggest transfering `purescript-uint` to `purescript-contrib` and possibly adding @jamesdbrock as a maintainer for it, if he wants to be one. If not then maybe `purescript-contrib` people themselve could maintain it. No need to add me in any role, I don't see myself being useful for the package in the foreseeable future.
So please let me know what you decide, whether you choose to include `purescript-uint` or not in `purescript-contrib`. I'll transfer it when requested by one of `purescript-contrib` public members. Thanks!

**Comment by JordanMartinez** [comment]
@thomashoneyman Any updates on this?

**Comment by thomashoneyman** [comment]
Sorry folks, I've been on a long vacation. I'm on board with transferring `uint` and `float32` into the contrib organization, and if we can make sure that `arraybuffer` doesn't have dependencies outside of the contrib / core organizations then I'm on board with that one as well.

**Comment by thomashoneyman** [comment]
@jamesdbrock I've invited you to the contributors organization, and you can now start transferring repositories. Let me know when you've initiated the transfer and I can approve them / update them for the contributors guidelines.

**Comment by thomashoneyman** [comment]
As far as `arraybuffer` goes, I'd want to remove the dependencies on `purescript-quickcheck-combinators` and `purescript-typelevel` before introducing the library to the Contributors organization. We need to ensure that we can update libraries for compiler releases, which means we need to control the dependency tree.

**Comment by zaquest** [comment]
Apparently one needs the right to create repositories in organization to transfer a repository to it. @jamesdbrock could you, please, remove your `purescript-uint` repository so that I could transfer it to you? And then you should be able to transfer it to `purescript-contrib`.

**Comment by jamesdbrock** [comment]
Ok @zaquest  I deleted jamesdbrock/purescipt-uint

**Comment by jamesdbrock** [comment]
My plan:

https://github.com/purescript/registry#name-squatting-and-reassigning-names

https://github.com/purescript/registry/issues/182

## purescript-uint

Transfer from zaquest to __purescript-contrib/purescript-uint__.

Reassign the same package name in the registry.

Add to package-sets.

(Later on, possibly add some of these instances?
https://github.com/athanclark/purescript-uint-instances/blob/master/src/Data/UInt/Instances.purs)

## purescript-float32

Transfer from athanclark to __purescript-contrib/purescript-float32__.

Reassign the same package name in the registry.

Upgrade to PureScript 0.14 and add to package-sets.

## purescript-arraybuffer

I will work on removing the non-purescript-contrib dependencies. Then we'll
see from there.

**Comment by jamesdbrock** [comment]
@jacereda has transferred __purescript-arraybuffer__ to __purescript-contrib__, thank you @jacereda .

Since __purescript-arraybuffer__ still has the non-contrib dependencies, should I transfer it to myself, or should I leave it in __purescript-contrib__ while we work on it?

**Comment by jamesdbrock** [comment]
I say we leave __purescript-arraybuffer__ in __purescript-contrib__ while we work on it.

**Comment by jamesdbrock** [comment]
It appears to me that `spago`, at least, can still download __purescript-arraybuffer__ through whatever redirect mechanism Github uses for transferred repositories. And probably everything else can still download it too.

https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/transferring-a-repository#whats-transferred-with-a-repository

So I think the current situation with __purescript-arraybuffer__ will not inconvenience anyone, and we can leave it as it is while we decide what to do.

**Comment by zaquest** [comment]
> It appears to me that `spago`, at least, can still download **purescript-arraybuffer** through whatever redirect mechanism Github uses for transferred repositories. And probably everything else can still download it too.

Yes, I think redirects is the main reason to use transfer over simply creating a fork, which could've been easier, but would've broken things in the end.

@jamesdbrock Transferred `purescript-uint` to you.

**Comment by jamesdbrock** [comment]
Sorry @zaquest the transfer expired, please transfer again?

**Comment by zaquest** [comment]
> Sorry @zaquest the transfer expired, please transfer again?

Aborted and requested transfer one more time.

**Comment by jamesdbrock** [comment]
@jacereda May I give you a “maintainer” badge on the __purescript-contrib/purescript-arraybuffer__ README? (along with myself, of course)

**Comment by jacereda** [comment]
Sure, I'm away from PS development these days, but I might come back at some point if I find a suitable project.

**Comment by jamesdbrock** [comment]
I guess I don't have permission for the Settings of https://github.com/purescript-contrib/purescript-arraybuffer repository? Would someone please set the default branch of the repository to __main__? @thomashoneyman @JordanMartinez

**Comment by JordanMartinez** [comment]
I've changed the default branch to `main`.

**Comment by jamesdbrock** [comment]
@athanclark would you please transfer https://github.com/athanclark/purescript-float32 to __jamesdbrock__?

**Comment by jamesdbrock** [comment]
In the unlikely case that @athanclark never receives my messages and never transfers the __purescript-float32__ repository and I fork the repo and add it to __purescript-contrib__, we’ll have to give it a new name. And maybe give the data type `Float32` a new name as well? Anyone want to suggest names? How about __purescript-float__ and `Float`?

**Comment by JordanMartinez** [comment]
I'm good with `purescript-float`. However, why not just keep the same `Float32` name? If anything, we could drop `purescript-float32` from the package set and replace it with `purescript-float` to minimize the breakage.

**Comment by athanclark** [comment]
@jamesdbrock Sorry for sleeping on this; I just got back from a vacation. I tried to transfer ownership to you, but you already have a fork with the same name as the original project. Maybe changing the name would be the best solution?

**Comment by jamesdbrock** [comment]
Hi @athanclark sorry I missed your transfer attempt! I renamed my fork of your repo to __jamesdbrock/purescript-float32-fork__ so that you would be able to transfer to me, but probably Github was still reserving __jamesdbrock/purescript-float32__ for redirects, which makes sense, I should have thought of that.

I have now entirely deleted the repo __jamesdbrock/purescript-float32-fork__, please try the transfer again?

**Comment by jamesdbrock** [comment]
> @jamesdbrock Sorry for sleeping on this; I just got back from a vacation.

(P.S. no problem, I hope you had a nice relaxing vacation!)

**Comment by athanclark** [comment]
Transferred!

**Comment by jamesdbrock** [comment]
Transfer complete, thank you again @athanclark !

**Comment by JordanMartinez** [comment]
Thanks!

**Comment by jamesdbrock** [comment]
I've completed the transfer of these three packages.

1. Ran `contrib-updater`.
2. Checked the https://github.com/purescript-contrib/governance/blob/main/library-guidelines.md
3. Upgraded to `purs` 0.14
4. Split out the testing dependencies into a separate Spago file.
5. Transferred in the __registry__.
6. Added to __package-sets__.
7. Published a new version on Pursuit.

* https://github.com/purescript-contrib/purescript-uint
* https://github.com/purescript-contrib/purescript-float32
* https://github.com/purescript-contrib/purescript-arraybuffer

I'm closing this issue .If there are any problems with what I've done, then let's discuss those on issues in each package's repo. Thanks again everyone!

**Comment by JordanMartinez** [comment]
Thanks for all your work @jamesdbrock!
